x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/route.ts

```diff
@@ -40,48 +37,18 @@ export const getInstalledIntegrationsRoute = (
         const fleet = ctx.securitySolution.getInternalFleetServices();
         const set = createInstalledIntegrationSet();
 
-        const packagePolicies = await fleet.packagePolicy.list(fleet.internalReadonlySoClient, {});
+        // Pulls all packages into memory just like the main fleet landing page
+        // No pagination support currently, so cannot batch this call
+        const allThePackages = await fleet.packages.getPackages();
+        allThePackages.forEach((fleetPackage) => {
+          set.addPackage(fleetPackage);
+        });
 
+        const packagePolicies = await fleet.packagePolicy.list(fleet.internalReadonlySoClient, {});
         packagePolicies.items.forEach((policy) => {
           set.addPackagePolicy(policy);
         });
 
-        const registryPackages = await initPromisePool({
-          concurrency: MAX_CONCURRENT_REQUESTS_TO_PACKAGE_REGISTRY,
-          items: set.getPackages(),
-          executor: async (packageInfo) => {
-            const registryPackage = await fleet.packages.getPackage(
-              packageInfo.package_name,
-              packageInfo.package_version
-            );
-            return registryPackage;
-          },
-        });
-
-        if (registryPackages.errors.length > 0) {
-          const errors = registryPackages.errors.map(({ error, item }) => {
-            return {
-              error,
-              packageId: `${item.package_name}@${item.package_version}`,
-            };
-          });
-
-          const packages = errors.map((e) => e.packageId).join(', ');
-          logger.error(
-            `Unable to retrieve installed integrations. Error fetching packages from registry: ${packages}.`
-          );
-
-          errors.forEach(({ error, packageId }) => {
-            const logMessage = `Error fetching package info from registry for ${packageId}`;
-            const logReason = error instanceof Error ? error.message : String(error);
-            logger.debug(`${logMessage}. ${logReason}`);
-          });
-        }
-
-        registryPackages.results.forEach(({ result }) => {
-          set.addRegistryPackage(result.packageInfo);
-        });
```

Comment:
Since package policies are available in each package of `const allThePackages = await fleet.packages.getPackages()`, we can stop fetching all the installed packages one-by-one. That reduces the latency of a cold request from several seconds to several hundred ms.

Reply:
Ah yeah! Nice catch, I totally missed that the policies were in there too. <img width="16" src="https://user-images.githubusercontent.com/2946766/141219243-86619f7f-c300-4355-96f5-316d70aa6a0e.png" />